napari/_qt/dialogs/qt_plugin_dialog.py

```diff
@@ -374,34 +594,37 @@ def handle_action(
                 if hasattr(item, 'latest_version'):
                     pkg_name += f"=={item.latest_version}"
 
-                widget.set_busy(trans._("updating..."), update)
+                widget.set_busy(trans._("updating..."), action_name, update)
                 widget.action_button.setDisabled(True)
             else:
-                widget.set_busy(trans._("installing..."), update)
+                if version:
+                    pkg_name += f"=={item.widget.version_choice_dropdown.currentText()}"
+                widget.set_busy(trans._("installing..."), action_name, update)
 
             job_id = self.installer.install(
                 tool=tool,
                 pkgs=[pkg_name],
                 # origins="TODO",
+                upgrade=update,
             )
-            widget.setProperty("current_job_id", job_id)
             if self._warn_dialog:
                 self._warn_dialog.exec_()
             self.scrollToTop()
         elif action_name == InstallerActions.UNINSTALL:
-            widget.set_busy(trans._("uninstalling..."), update)
+            widget.set_busy(trans._("uninstalling..."), action_name, False)
```

Comment:
It is better to use named argument when using bool
```suggestion
            widget.set_busy(trans._("uninstalling..."), action_name, update=False)
```